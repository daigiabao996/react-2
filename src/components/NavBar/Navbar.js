import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link, useNavigate } from "react-router-dom";
import { handleLogoutAuthedUser } from "../../actions/authedUser";

const Navbar = () => {
  const authUser = useSelector((state) => state.authUser);
  const navigate = useNavigate();
  const dispatch = useDispatch();
  const logoutHandle = () => {
    dispatch(handleLogoutAuthedUser());
    navigate("/login");
  };
  return (
    <div className="navbar bg-base-100">
      {authUser && (
        <>
          <div className="flex-1">
            <ul data-testid="navbuttons" className="menu menu-horizontal px-1">
              <li>
                <Link data-testid="home-link" to={"/"}>
                  Home
                </Link>
              </li>
              <li>
                <Link data-testid="new-poll-link" to={"/add"}>
                  New Poll
                </Link>
              </li>
              <li>
                <Link data-testid="LeaderBoard-link" to={"/LeaderBoard"}>
                  LeaderBoard
                </Link>
              </li>
            </ul>
          </div>
          <div className="flex-2">
            <span className="mr-2">
              {authUser.avatarURL ? (
                <img
                  src={authUser.avatarURL}
                  alt={authUser.name}
                  className="w-10 h-10 rounded-full"
                />
              ) : (
                <img
                  src={`https://avatar.iran.liara.run/username?username=${authUser.name}`}
                  alt={authUser.name}
                  className="w-10 h-10 rounded-full"
                />
              )}
            </span>
            <h2 data-testid="authUserName" className="text-center text-xl mr-4">
              {authUser.name}
            </h2>
            <button data-testid="logout-link" onClick={logoutHandle}>
              Logout
            </button>
          </div>
        </>
      )}
    </div>
  );
};

export default Navbar;
