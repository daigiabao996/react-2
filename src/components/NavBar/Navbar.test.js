import React from "react";
import Navbar from "./Navbar";
import reducer from "../../reducers";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { configureStore } from "@reduxjs/toolkit";
import { loginAuthedUser } from "../../actions/authedUser";

const store = configureStore({ reducer });

describe("Navbar", () => {
  test("should render the component", () => {
    store.dispatch(
      loginAuthedUser({
        username: "tylermcginnis",
        password: "",
      })
    );
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toBeDefined();
    expect(view).toMatchSnapshot();
  });

  test("should display all elements", () => {
    store.dispatch(
      loginAuthedUser({
        username: "tylermcginnis",
        password: "",
      })
    );
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>
      </Provider>
    );

    const homeLinkElement = screen.getByTestId("home-link");
    const newPollLinkElement = screen.getByTestId("new-poll-link");
    const LeaderBoardLinkElement = screen.getByTestId("LeaderBoard-link");
    const logoutLinkElement = screen.getByTestId("logout-link");

    expect(homeLinkElement.textContent).toBe("Home");
    expect(newPollLinkElement.textContent).toBe("New Poll");
    expect(LeaderBoardLinkElement.textContent).toBe("LeaderBoard");
    expect(logoutLinkElement.textContent).toBe("Logout");
  });
});
