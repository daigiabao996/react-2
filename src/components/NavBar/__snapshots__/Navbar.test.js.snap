// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navbar bg-base-100"
      >
        <div
          class="flex-1"
        >
          <ul
            class="menu menu-horizontal px-1"
            data-testid="navbuttons"
          >
            <li>
              <a
                data-testid="home-link"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                data-testid="new-poll-link"
                href="/add"
              >
                New Poll
              </a>
            </li>
            <li>
              <a
                data-testid="LeaderBoard-link"
                href="/LeaderBoard"
              >
                LeaderBoard
              </a>
            </li>
          </ul>
        </div>
        <div
          class="flex-2"
        >
          <span
            class="mr-2"
          >
            <img
              class="w-10 h-10 rounded-full"
              src="https://avatar.iran.liara.run/username?username=undefined"
            />
          </span>
          <h2
            class="text-center text-xl mr-4"
            data-testid="authUserName"
          />
          <button
            data-testid="logout-link"
          >
            Logout
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="navbar bg-base-100"
    >
      <div
        class="flex-1"
      >
        <ul
          class="menu menu-horizontal px-1"
          data-testid="navbuttons"
        >
          <li>
            <a
              data-testid="home-link"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              data-testid="new-poll-link"
              href="/add"
            >
              New Poll
            </a>
          </li>
          <li>
            <a
              data-testid="LeaderBoard-link"
              href="/LeaderBoard"
            >
              LeaderBoard
            </a>
          </li>
        </ul>
      </div>
      <div
        class="flex-2"
      >
        <span
          class="mr-2"
        >
          <img
            class="w-10 h-10 rounded-full"
            src="https://avatar.iran.liara.run/username?username=undefined"
          />
        </span>
        <h2
          class="text-center text-xl mr-4"
          data-testid="authUserName"
        />
        <button
          data-testid="logout-link"
        >
          Logout
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
